src/main/java/com/davidthomasbernal/stardict/Dictionary.java

```diff
@@ -71,22 +71,15 @@ public static Dictionary fromIfo(String path, boolean tolerateInfoMismatch) thro
             throw new IllegalArgumentException("Idx file does not exist");
         }
 
-        Reader ifoReader = null;
-        DictionaryInfo dictionaryInfo = null;
-
-        try {
-            ifoReader = new InputStreamReader(new FileInputStream(ifo), StandardCharsets.UTF_8);
+        DictionaryInfo dictionaryInfo;
+        try (Reader ifoReader = new InputStreamReader(new FileInputStream(ifo), StandardCharsets.UTF_8)) {
 
             IfoParser ifoParser = new IfoParser();
             dictionaryInfo = ifoParser.parse(ifoReader);
-        } finally {
-            if (ifoReader != null) {
-                ifoReader.close();
-            }
         }
 
         BufferedInputStream stream = null;
-        DictionaryIndex dictionaryIndex = null;
```

Comment:
Let's keep this initialization please - clearer.

Reply:
It may be clearer but less safier: it's dangerous practice to set value multiple times.
For example in Scala, Kotlin, JS it even splitted to keywords: `var` and `val` - val is `final` so can't be setted twice